app/Services/PolygonService.php

```diff
@@ -212,24 +215,25 @@ protected function validateSitePolygonProperties(string $polygonUuid, array $pro
      *
      * @return string UUID of resulting PolygonGeometry
      */
-    protected function transformAndStorePoints($geojson, $sitePolygonProperties): string
+    protected function transformAndStorePoints($geojson, $sitePolygonProperties): array
     {
-        $pointUuids = [];
-        foreach ($geojson['features'] as $feature) {
-            $pointUuids[] = $this->insertSinglePoint($feature);
+        foreach ($geojson['features'] as &$feature) {
+            $currentPointUUID = $this->insertSinglePoint($feature);
+            $feature['properties']['point_id'] = $currentPointUUID;
         }
 
         $properties = $sitePolygonProperties;
+        $mainSiteID = '';
         foreach (self::POINT_PROPERTIES as $property) {
             $properties[$property] = collect(data_get($geojson, "features.*.properties.$property"))->filter()->first();
+            if ($property === 'site_id') {
+                $mainSiteID = $properties[$property];
+            }
         }
 
-        // TODO:
-        //  * transform points into a polygon
-        //  * Insert the polygon into PolygonGeometry
-        //  * Create the SitePolygon using the data in $properties (including $properties['site_id'] to identify the site)
-        //  * Return the PolygonGeometry's real UUID instead of this fake return
+        $polygonsGeojson = App::make(PythonService::class)->voronoiTransformation($geojson);
```

Comment:
One more thing; please throw an exception if `$polygonsGeojson` is null. That's causing an error in the unit test, but is a legit problem that could happen in prod as well if something goes wrong. I'm working on fixing up the unit test now.

Reply:
Now it throws the exception now.